test/GithubApi.Gist.Fetch.test.js

```diff
@@ -0,0 +1,67 @@
+const { expect } = require('chai');
+const { StatusCodes } = require('http-status-codes');
+require('isomorphic-fetch');
+
+const headers = {
+  Authorization: `token ${process.env.ACCESS_TOKEN}`
+};
+
+const baseUrl = 'https://api.github.com';
+const gistCreated = {};
+
+describe('FETCH - Consume DELETE method and non-existing resource', () => {
+  it('Should create a gist and must be verifying its data', async () => {
+    const gistBody = {
+      description: 'Example of a gist',
+      public: false,
+      files: {
+        'README.md': {
+          content: 'Hello World by Diegocortes15'
+        }
+      }
+    };
+
+    const response = await fetch(`${baseUrl}/gists`, {
+      method: 'POST',
+      headers,
+      body: JSON.stringify(gistBody)
+    });
+
+    expect(response.status).to.equal(StatusCodes.CREATED);
+
+    const data = await response.json();
+
+    gistCreated.url = data.url;
+
+    expect(data).to.containSubset(gistBody);
+  });
```

Comment:
```suggestion
  const gistBody = {
      description: 'Example of a gist',
      public: false,
      files: {
        'README.md': {
          content: 'Hello World by Diegocortes15'
        }
      }
    };
  let response;
  let data;
  before (async () = {
    response = await fetch(`${baseUrl}/gists`, {
      method: 'POST',
      headers,
      body: JSON.stringify(gistBody)
    });
  })

  it('Should create a gist and must be verifying its data', async () => {
    expect(response.status).to.equal(StatusCodes.CREATED);

    data = await response.json();
    expect(data).to.containSubset(gistBody);
  });
```

This is to reuse the responses on other tests without breaking the rule of the tests being independent.

Also avoid to mutate objects used to make requests, and better use the properties from the responses